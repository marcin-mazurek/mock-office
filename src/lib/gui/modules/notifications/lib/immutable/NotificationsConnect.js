import { connect } from 'react-redux';
import Notifications from '../Notifications';
import { newNotificationDisplayedAction, notificationsClickedAction } from '../actions';
import { allNotificationsSelector } from './selectors';

const mapStateToProps = state => ({
  notifications: allNotificationsSelector(state)
});
const mapDispatchToProps = {
  onNotificationClick: notificationsClickedAction,
  onNewNotificationDisplayed: newNotificationDisplayedAction
};
export default connect(mapStateToProps, mapDispatchToProps)(Notifications);
