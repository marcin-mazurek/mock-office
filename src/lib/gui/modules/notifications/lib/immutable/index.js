export * from './selectors';
export { default as NotificationsConnect } from './NotificationsConnect';
