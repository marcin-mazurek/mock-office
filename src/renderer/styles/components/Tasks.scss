.tasks {
  @extend %list-reset;
  display: flex;
  flex-direction: column;
  width: 100%;
}

.tasks__header {
  margin: 10px 0;
  font-weight: 500;
}

.tasks__item {
  margin-bottom: 10px;
}

.task {
  border: 1px solid #EEE;
  flex: none;
  padding: 5px;
  width: 100%;
  display: flex;
  align-items: center;
}

.task-details__row {
  margin: 10px;
}

.task__remove-button {
  @extend %button-reset;
  font-size: 16px;
  color: #999;
  margin-right: 10px;
}
