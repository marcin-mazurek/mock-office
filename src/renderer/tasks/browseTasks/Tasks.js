import React from 'react';
import { connect } from 'react-redux';
import { getTask } from '../selectors';
import { init } from '../removeTask/actions';
import { getQueueTaskIds } from '../../queues/selectors';

export const Task = ({ id, taskPayload, queueId, remove }) =>
  <div className="task">
    <div>{JSON.stringify(taskPayload)}</div>
    <button onClick={() => remove(queueId, id)}>Remove</button>
  </div>;

Task.propTypes = {
  id: React.PropTypes.string.isRequired,
  taskPayload: React.PropTypes.shape({}).isRequired,
  queueId: React.PropTypes.string.isRequired,
  remove: React.PropTypes.func.isRequired
};

const taskMapStateToProps = (initialState, ownProps) => state => ({
  taskPayload: getTask(state, ownProps).taskPayload
});

const taskMapDispatchToProps = {
  remove: init
};

export const TaskConnect = connect(taskMapStateToProps, taskMapDispatchToProps)(Task);

export const Tasks = ({ taskIds, queueId }) => (
  <ul className="tasks">
    {
      taskIds.map(taskId =>
        <li className="tasks__item" key={taskId}>
          <TaskConnect queueId={queueId} id={taskId} />
        </li>
      )
    }
  </ul>
);

Tasks.propTypes = {
  taskIds: React.PropTypes.shape({}).isRequired,
  queueId: React.PropTypes.string.isRequired
};

const mapStateToProps = (state, ownProps) => ({
  taskIds: getQueueTaskIds(ownProps.queueId, state)
});

export default connect(mapStateToProps)(Tasks);
