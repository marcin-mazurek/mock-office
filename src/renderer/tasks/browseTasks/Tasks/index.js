import React from 'react';
import { connect } from 'react-redux';
import { getTask } from '../../selectors';
import { init } from '../../removeTask/actions';
import { getQueueTaskIds } from '../../../queues/selectors';

const Tasks = ({ tasks, queueId, remove }) => (
  <ul className="tasks">
    {
      tasks.map(task => (
        <li className="tasks__item" key={task.id}>
          <div className="task">
            <div>{JSON.stringify(task.taskPayload)}</div>
            <button onClick={() => remove(queueId, task.id)}>Remove</button>
          </div>
        </li>
      ))
    }
  </ul>
);

Tasks.propTypes = {
  tasks: React.PropTypes.shape({}).isRequired,
  queueId: React.PropTypes.string.isRequired,
  remove: React.PropTypes.func.isRequired
};

const mapStateToProps = (state, ownProps) => ({
  tasks: getQueueTaskIds(ownProps.queueId, state).map(taskId => getTask(state, taskId))
});

const mapDispatchToProps = {
  remove: init
};

export default connect(mapStateToProps, mapDispatchToProps)(Tasks);
