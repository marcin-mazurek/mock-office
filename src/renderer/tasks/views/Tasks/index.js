import React from 'react';
import { connect } from 'react-redux';
import { getTask } from '../../selectors';
import { init } from '../../remove/actions';

const TaskComponent = ({ task, remove, queueId, taskId }) => (
  <div className="task">
    <div>{JSON.stringify(task)}</div>
    <button onClick={() => remove(queueId, taskId)}>Remove</button>
  </div>
);

TaskComponent.propTypes = {
  taskId: React.PropTypes.string.isRequired,
  task: React.PropTypes.shape({}).isRequired,
  remove: React.PropTypes.func.isRequired,
  queueId: React.PropTypes.string.isRequired
};

const mapStateToProps = (state, ownProps) => ({
  task: getTask(state, ownProps.taskId)
});

const mapDispatchToProps = {
  remove: init
};

const Task = connect(mapStateToProps, mapDispatchToProps)(TaskComponent);

const Tasks = ({ tasksIds, queueId }) => (
  <ul className="tasks">
    {
      tasksIds.map(id => (
        <li className="tasks__item" key={id}>
          <Task taskId={id} queueId={queueId} />
        </li>
      ))
    }
  </ul>
);

Tasks.propTypes = {
  tasksIds: React.PropTypes.shape({}).isRequired,
  queueId: React.PropTypes.string.isRequired
};

export default Tasks;
