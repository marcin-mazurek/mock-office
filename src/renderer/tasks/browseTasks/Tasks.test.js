import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { List } from 'immutable';
import { Tasks } from './Tasks';

test('Tasks should render', () => {
  const props = {
    tasks: new List(),
    queueId: 'some id',
    remove: () => {}
  };
  const wrapper = shallow(<Tasks {...props} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
