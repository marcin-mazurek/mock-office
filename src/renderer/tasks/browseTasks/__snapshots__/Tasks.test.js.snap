exports[`Tasks default snapshot 1`] = `
<ul
  className="tasks" />
`;
